### ISSUE TITLE
mapstructure.DecoderConfig.Squash support config in Options()

### ISSUE THREAD

**Issue by RelicOfTesla** [gookit/config#45]
```go
cfgLoader := config.NewWithOptions("", func(options *config.Options) {
    options.TagName = "json"
    options.Squash  = true
})

type Test1 struct {
    B int `json:"b"`
}
type Test2 struct {
    Test1
    C int `json:"c"`
}
cfg := &Test2{}

cfgLoader.BindStruct("", cfg)
```
```go
mapConf := &mapstructure.DecoderConfig{
		Metadata: nil,
		Result:   dst,
		TagName:  c.opts.TagName,
                Squash :  c.opts.Squash ,                   // support embedded structs
		// will auto convert string to int/uint
		WeaklyTypedInput: true,
	}

```

**Comment by RelicOfTesla** [comment]
or 
```
cfgLoader := config.NewWithOptions("", func(options *config.Options) {
    options.DefaultStructDecoderConfig = &mapstructure.DecoderConfig{
         DecodeHook : func(){}....

    }
})
```

**Comment by inhere** [comment]
hi @RelicOfTesla 
In my test, `Square` does not affect the structure data embedded in the binding

https://github.com/gookit/config/blob/e7902e131a1cf3dca3fc513b82c85bddc5bdcc7c/export_test.go#L141-L184

![image](https://user-images.githubusercontent.com/5302062/112340357-6d198100-8cfb-11eb-83b4-f57dbdfa7bf7.png)

https://github.com/gookit/config/blob/e7902e131a1cf3dca3fc513b82c85bddc5bdcc7c/export_test.go#L186-L230

![image](https://user-images.githubusercontent.com/5302062/112340177-49563b00-8cfb-11eb-8ae0-4618369daffe.png)

**Comment by RelicOfTesla** [comment]
I'm test success? github.com/mitchellh/mapstructure@1.4.1 support not ptr embedded in the binding...

```go
mapConf := &mapstructure.DecoderConfig{
		Metadata: nil,
		Result:   dst,
		TagName:  c.opts.TagName,
                Squash :  true ,                   // force code... support embedded structs
		// will auto convert string to int/uint
		WeaklyTypedInput: true,
	}
```

**Comment by RelicOfTesla** [comment]
your test json wrong...
```json
json = {
 "c": "12",
 "b": "34"
}
```

**Comment by inhere** [comment]
now, has been add `Options.DecoderConfig`, can custom the `mapstructure.DecoderConfig`

eg:

```go
loader := NewWithOptions("test", func(options *Options) { 
 		options.DecoderConfig.TagName = "json" 
 		options.DecoderConfig.Squash = true 
 	}) 
```

**Comment by inhere** [comment]
Released `v2.0.23`
